src/com/lilithsthrone/rendering/Pattern.java

```diff
@@ -0,0 +1,111 @@
+package com.lilithsthrone.rendering;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Pattern {
+	
+	private static Map<String, Pattern> allPatterns;
+	
+	private String name;
+	private String SVGString; // TODO ADD THIS
+	
+	static {
+		allPatterns = new HashMap<>();
+		
+		File dir = new File("res/patterns");
+		
+		allPatterns.put("none", new Pattern("none")); // Adding empty pattern
+		
+		if(dir.exists()) {
+			FilenameFilter textFilter = new FilenameFilter() {
+				public boolean accept(File dir, String name) {
+					String lowercaseName = name.toLowerCase();
+					if (lowercaseName.endsWith(".svg")) {
```

Comment:
It might be possible to turn this entire thing into a lambda; dir.listFiles((path, name) -> name.endsWith(".svg")) - see EnchantmentDialogue (around line 580) for example

Reply:
I'll be honest, I was copypasting some other code in there because I don't know Java's file IO libraries well enough

Reply:
From where exactly? Seems like that should be cleaned up